### PR TITLE
Fix CI after foundry update

### PR DESCRIPTION
**Pull request by MartinquaXD**
# Description
https://github.com/foundry-rs/foundry/issues/8035 changed `anvil` to compute the gas price differently. This seems like a breaking change since under some conditions the new logic computes a different gas price than the original one.

This broke 2 of our tests which both involve waiting for the system to flush out cached liquidity data to generate new quotes. For `e2e::e2e quoting::local_node_test` it was enough to just bump the `--base-fee` to `1` (without breaking any other test) and for `e2e::e2e protocol_fee::local_node_combined_protocol_fees` I needed to make the condition to detect flushed liquidity stricter.
Instead of waiting for any change in quotes (which are now triggered too early because of tiny changes in the gas price) we now wait for the quote to be at least twice as good as the old quote.

# Changes
- bump `--base-fee` to `1` in `anvil` args
- stricter synchronization logic for `e2e::e2e protocol_fee::local_node_combined_protocol_fees`
- use nightly version of foundry again in CI

### CODE REVIEW

File: crates/e2e/src/setup/mod.rs [hunk 1]
```diff
@@ -74,7 +74,7 @@ fn with_default_filters<T>(custom_filters: impl IntoIterator<Item = T>) -> Vec<S
 where
     T: AsRef<str>,
 {
-    let mut default_filters: Vec<_> = DEFAULT_FILTERS.into_iter().map(String::from).collect();
+    let mut default_filters: Vec<_> = DEFAULT_FILTERS.iter().map(|s| s.to_string()).collect();
```
Comment (Mateo-mro) [inline crates/e2e/src/setup/mod.rs:77]: why this change? :thinking:

File: crates/e2e/tests/e2e/protocol_fee.rs [hunk 2]
```diff
@@ -231,7 +231,7 @@ async fn combined_protocol_fees(web3: Web3) {
         )
         .await
         .unwrap();
-        new_market_order_quote.quote.buy_amount != market_quote_before.quote.buy_amount
+        new_market_order_quote.quote.buy_amount > market_quote_before.quote.buy_amount * 2
```
Comment (Mateo-mro) [inline crates/e2e/tests/e2e/protocol_fee.rs:234]: from where is coming the `2`?

File: crates/e2e/src/setup/mod.rs (hunk 1 above)
Comment (MartinquaXD) [reply crates/e2e/src/setup/mod.rs:77]: Whenever you want to add or remove a log filter from that list you also have to update its type because it's a statically sized array. With this change I convert the thing into a fat pointer which knows its own length so you don't have to jump through these hoops anymore. Technically not needed for this PR but it annoyed me yet again so I just wanted to finally fix it.

File: crates/e2e/tests/e2e/protocol_fee.rs (hunk 2 above)
Comment (MartinquaXD) [reply crates/e2e/tests/e2e/protocol_fee.rs:234]: This `2` isn't really based in anything. The actual new price is orders of magnitude bigger and the difference in buy amount due to gas prices is way smaller than that. I just wanted to make the smallest possible change without having to deal with floating point math.
If you feel like it's helpful or necessary I can add a comment but I wouldn't want to make this a constants since it's not relevant for anything else.

**Review by squadgazzz — APPROVED** [review]
LG, also a comment on the `new_market_order_quote.quote.buy_amount > market_quote_before.quote.buy_amount * 2` would be helpful.

File: crates/e2e/src/setup/mod.rs (hunk 1 above)
Comment (Mateo-mro) [reply crates/e2e/src/setup/mod.rs:77]: oh, I understand your point, got it! :ok_hand: thanks for the explanation!

File: crates/e2e/tests/e2e/protocol_fee.rs (hunk 2 above)
Comment (Mateo-mro) [reply crates/e2e/tests/e2e/protocol_fee.rs:234]: Thanks a lot! :pray: please, could you add a comment for it?

**Review by Mateo-mro — APPROVED** [review]

File: crates/e2e/tests/e2e/protocol_fee.rs (hunk 2 above)
Comment (MartinquaXD) [reply crates/e2e/tests/e2e/protocol_fee.rs:237]: Done

**Review by fleupold — APPROVED** [review]
